### ISSUE TITLE
ObjectDisposedException when disconnecting from RpcServer

### ISSUE THREAD

**Issue by esskar** [SciTechSoftware/SciTech.Rpc#3]
Hello, thanks for the library.
I am currently running standalone RpcServer as a NT service, to schedule jobs that are created from an ASP.NET Service. So that ASP.NET Service acts as a client to the RPC Service. Now, when the ASP.NET Service gets updated, i see the following exception on the RPC Service

```
System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'RpcPipeline'.
   at SciTech.Rpc.Lightweight.Internal.RpcPipeline.<EndWriteAsync>g__FinalizeWriteAsync|25_2(RpcPipeline self, BufferWriterStream frameWriter)
   at SciTech.Rpc.Lightweight.Internal.RpcPipeline.EndWriteAsync(WriteState& writeState, Boolean throwOnError)
   at SciTech.Rpc.Lightweight.Server.LightweightRpcServer.ClientPipeline.WriteErrorResponseAsync(Int32 messageId, String operationName, RpcError rpcError, IRpcSerializer serializer)
   at SciTech.Rpc.Lightweight.Server.LightweightRpcServer.ClientPipeline.<>c__DisplayClass23_0.<<HandleAsyncOperation>g__HandleCompletedAsyncOperation|0>d.MoveNext()
```

Is there a way to handle that better?

**Comment by ASuurkuusk** [comment]
Thank you for reporting this issue.

Is the ObjectDisposedException handled by the RPC library, or does it propagate out as an unhandled exception? 

We have also noticed that you can get ObjectDisposedExceptions in active service calls when a client gets disconnected, which is most evident if there are many streaming calls or active event handlers. These exceptions are handled by the RPC library, but get detected by the debugger which is a bit annoying. 

After your issue report I did look into this, and fixed the problems we are seeing. The fix is included in the latest Nuget packages, v1.0.0-beta.25 (and pushed to master). Can you upgrade to this version and see if it resolves your exception as well?

**Comment by esskar** [comment]
Yes, this fixes the issue. Thanks for the quick reply and fix.
